biogrid/libs/biogrid-simulator/src/lib/building/building.ts

```diff
@@ -0,0 +1,47 @@
+import { EnergyUser } from '@biogrid/grid-simulator';
+
+/**
+ * A structure such as a building or house which uses energy to operate.
+ */
+export class Building implements EnergyUser {
+
+  private energyInJoules: number;
+  private buildingId: number;
+  /** The battery storage for the building. */
+  battery: unknown;
+
+  /**
+   * @param {number} energy Amount of energy the building will have in joules.
+   */
+  constructor(energy: number) {
+    this.energyInJoules = energy;
+    this.buildingId = -1;
```

Comment:
nit: can you leave a comment saying that -1 here means the id is essentially "unset", and will be set later